glass/shells.py

```diff
@@ -195,16 +197,31 @@ class RadialWindow(NamedTuple):
     zeff
         Effective redshift of the window.
 
-    Methods
-    -------
-    _replace
-        Create a new instance with changed attribute values.
-
     """
 
     za: NDArray[np.float64]
     wa: NDArray[np.float64]
-    zeff: float = 0
+    zeff: float = dataclasses.field(default=math.nan)
+
+    def __post_init__(self) -> None:
+        """Magic method to calculate the effective redshift if not given."""
+        if math.isnan(self.zeff):
+            object.__setattr__(self, "zeff", self._calculate_zeff())
+
+    def _calculate_zeff(self) -> float:
+        """Calculate ``zeff`` if not given.
+
+        Returns
+        -------
+            The effective redshift depending on the size of ``za``.
+
+        """
+        if self.za.size > 0:
+            return np.trapezoid(  # type: ignore[return-value]
+                self.za * self.wa,
+                self.za,
+            ) / np.trapezoid(self.wa, self.za)
+        return 0.0
```

Comment:
Maybe we should return `math.nan` here instead of 0?

Reply:
Changed